app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js

```diff
@@ -0,0 +1,347 @@
+  if (setup.withAuth) {
+    await setState(ClientFunction)('fakeToken', authTokenInSession);
+  }
+  if (setup.getRoute) {
+    mocks(setup.mockData);
+    await setState(ClientFunction)('fakeToken', authTokenInSession);
+    await setState(ClientFunction)(sessionKeys.selectedRecipientId, selectedRecipientIdInSession);
+    await setState(ClientFunction)(sessionKeys.selectedRecipientName, selectedRecipientNameInSession);
+    await setState(ClientFunction)(sessionKeys.selectedItemId, itemIdInSession);
+    await setState(ClientFunction)(sessionKeys.selectedItemName, itemNameInSession);
+    await setState(ClientFunction)(sessionKeys.selectedPriceId, selectedPriceIdInSession);
+    await setState(ClientFunction)(sessionKeys.catalogueSolutionId, catalogueSolutionIdInSession);
+    await setState(ClientFunction)(sessionKeys.plannedDeliveryDate, deliveryDateInSession);
+    await setState(ClientFunction)(sessionKeys.recipients, recipientsInSession);
+    await setState(ClientFunction)(sessionKeys.selectedRecipients, selectedRecipientsInSession);
+  }
+  if (setup.postRoute) {
+    await setState(ClientFunction)(sessionKeys.orderItemPageData, orderItemPageDataInSession);
+  }
+};
+
+fixture('Catalogue-solutions - flat ondemand - general')
+  .page('http://localhost:1234/order/some-fake-page')
+  .afterEach(async (t) => {
+    await nockAndErrorCheck(nock, t);
+  });
+
+test('should render the solution table headings', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const solutionTable = Selector('div[data-test-id="solution-table"]');
+  const solutionNameColumnHeading = solutionTable.find('[data-test-id="column-heading-0"]');
+  const practiceSizeColumnHeading = solutionTable.find('[data-test-id="column-heading-1"]');
+  const dateColumnHeading = solutionTable.find('[data-test-id="column-heading-2"]');
+
+  await t
+    .expect(solutionTable.exists).ok()
+    .expect(solutionNameColumnHeading.exists).ok()
+    .expect(await extractInnerText(solutionNameColumnHeading)).eql(content.solutionTable.columnInfo[0].data)
+    .expect(practiceSizeColumnHeading.exists).ok()
+    .expect(await extractInnerText(practiceSizeColumnHeading)).eql(content.solutionTable.columnInfo[1].data)
+    .expect(dateColumnHeading.exists).ok()
+    .expect(await extractInnerText(dateColumnHeading)).eql(`${content.solutionTable.columnInfo[2].data}\n${content.solutionTable.columnInfo[2].additionalAdvice}`);
+});
+
+test('should render the solution table content', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const table = Selector('div[data-test-id="solution-table"]');
+  const row = table.find('[data-test-id="table-row-0"]');
+  const solutionName = row.find('div[data-test-id="recipient-name-code-recipient"]');
+  const quantityInput = row.find('[data-test-id="question-quantity"] input');
+  const practiceSizeExpandableSection = row.find('[data-test-id="view-section-input-id-practice"]');
```

Comment:
Similar to above, we should rename this `const` to `quantityExpandableSection`.

Reply:
Done